src/lib/utils.ts

```diff
@@ -75,13 +81,56 @@ export const debug = (message: string, details?: any): void => {
   }
 };
 
-export const isIssuedTimeWithinWindow = (expiresAtMs: number, expiresInMs = 691199000 /* 8 days */, windowMs = 1680 * 1000 /* default: 28 minutes */): boolean => {
+/**
+ * Determine if a token was issued within a given timeframe window. Example may be:
+ *  a token is received (either from localStorage or some other way) and you need
+ *  to be able to tell if it was issued within the last 10 minutes. Use this function
+ *  as follows:
+ * ``` ts
+ * const tokenExpiryTime = 1624611600000; // "2021-06-25 09:00.000Z"
+ * const tokenExpiresIn  = 86400000; // 1 day: meaning token was issuedAt "2021-06-24 09:00.000Z"
+ * const timeframe = 600000; // 10 minutes
+ * const startTime = 1624525260000; // "2021-06-24 09:01.000Z": 1 minute after issued time
+ *
+ * const willBeTrue = isIssuedTimeWithinTimeframe(
+ *  tokenExpiryTime,
+ *  tokenExpiresIn,
+ *  timeframe,
+ *  startTime
+ * );
+ * ```
+ * @param expiresAtMs epoch time (in milliseconds) for when the token will expire
+ * @param expiresInMs milliseconds for how long the token is valid for. Default `691199000`
+ * @param timeframe timeframe (in milliseconds) to check if the token was issued within. Default `1680000`
+ * @param startTime time to count `timeframe` back from (in epoch time). Default is `Date.now()`
+ * @returns boolean of whether the token was issued within the given timeframe (from now)
+ */
+export const isIssuedTimeWithinTimeframe = (
+  expiresAtMs: number,
+  expiresInMs = 691199000 /* 8 days */,
+  timeframe = 1680 * 1000 /* default: 28 minutes */,
+  startTime = Date.now()
+): boolean => {
   const issuedAt = tokenWasIssuedAt(expiresAtMs, expiresInMs);
 
   /* if current time is within the window from issuedAt */
-  return Date.now() - issuedAt < windowMs;
+  return startTime - issuedAt < timeframe;
 };
 
+/**
+ * @deprecated use `isIssuedTimeWithinTimeframe()`
+ */
+export const isIssuedTimeWithinWindow = (expiresAtMs: number, expiresInMs?: number, timeframe?: number): boolean => {
```

Comment:
Deprecated this since the function name wasn't super clear (I know, only one release and I already have deprecation lol). `isIssuedTimeWithinTimeframe()` above is more clear and accepts a `startTime` param. This will probably only be used by volt, but since we expose it, I thought it should be named well and have good docs.